### PR TITLE
[DON'T MERGE]Added api to fetch default evaluation config

### PR DESCRIPTION
**Pull request by lu-ohai**
### Added api to fetch default evaluation config

- For offline model, read `evaluation_config.json` from service model object storage path.
- For deployed model, use:
-- Job shapes from `list_job_shapes`: 1). use the same info for GPU and non-flex CPU shape 2). use the default ocpus and memory in gbs for flex CPU shape.
-- Default model params
-- Default block storage size

### Notebook
- Deployed model
```
{'shape': {'VM.Standard2.1': {'block_storage_size': 100,
   'ocpus': 1,
   'memory_in_gbs': 15},
  'VM.Standard2.2': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 30},
  'VM.Standard2.4': {'block_storage_size': 100,
   'ocpus': 4,
   'memory_in_gbs': 30},
  'VM.Standard2.8': {'block_storage_size': 100,
   'ocpus': 8,
   'memory_in_gbs': 120},
  'VM.Standard2.16': {'block_storage_size': 100,
   'ocpus': 16,
   'memory_in_gbs': 240},
  'VM.Standard2.24': {'block_storage_size': 100,
   'ocpus': 24,
   'memory_in_gbs': 320},
  'VM.Standard.E3.Flex': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 32},
  'VM.Standard.E4.Flex': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 32},
  'VM.Standard3.Flex': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 32},
  'VM.Optimized3.Flex': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 32},
  'VM.GPU2.1': {'block_storage_size': 100, 'ocpus': 12, 'memory_in_gbs': 72},
  'VM.GPU3.1': {'block_storage_size': 100, 'ocpus': 6, 'memory_in_gbs': 90},
  'VM.GPU3.2': {'block_storage_size': 100, 'ocpus': 12, 'memory_in_gbs': 180},
  'VM.GPU3.4': {'block_storage_size': 100, 'ocpus': 24, 'memory_in_gbs': 360},
  'BM.GPU2.2': {'block_storage_size': 100, 'ocpus': 28, 'memory_in_gbs': 192},
  'BM.GPU.T1.2': {'block_storage_size': 100,
   'ocpus': 32,
   'memory_in_gbs': 1024},
  'BM.GPU3.8': {'block_storage_size': 100, 'ocpus': 52, 'memory_in_gbs': 768},
  'BM.GPU4.8': {'block_storage_size': 100, 'ocpus': 64, 'memory_in_gbs': 2048},
  'BM.GPU.A10.4': {'block_storage_size': 100,
   'ocpus': 64,
   'memory_in_gbs': 1024},
  'VM.GPU.A10.1': {'block_storage_size': 100,
   'ocpus': 15,
   'memory_in_gbs': 240},
  'VM.GPU.A10.2': {'block_storage_size': 100,
   'ocpus': 30,
   'memory_in_gbs': 480},
  'VM.Standard.A1.Flex': {'block_storage_size': 100,
   'ocpus': 2,
   'memory_in_gbs': 32}},
 'model_params': {'max_tokens': 500,
  'temperature': 0.7,
  'top_p': 1.0,
  'top_k': 50},
 'default': {'ocpus': 2, 'memory_in_gbs': 32, 'block_storage_size': 100}}
```

### CODE REVIEW

**Review by VipulMascarenhas — APPROVED** [review]
changes look good.

File: ads/aqua/utils.py [hunk 1]
```diff
@@ -385,3 +407,14 @@ def upload_file_to_os(
         auth=auth,
         force_overwrite=force_overwrite,
     )
+
+def load_default_aqua_config(artifact_path: str, **kwargs) -> dict:
+    config = json.loads(
+        read_file(file_path=artifact_path, **kwargs) or UNKNOWN_JSON_STR
+    )
+    if not config:
```
Comment (mingkang111) [inline ads/aqua/utils.py:415]: We can use `pydantic` to help with validating returned evaluation config (as well as other response) in the future.

**Review by mingkang111 — APPROVED** [review]

**Comment by mingkang111** [comment]
We currently don't have model with `evaluation_config.json` in artifact for our testing. I would suggest to providing a test that proving the response return as expected when `evaluation_config.json` is in artifact.

File: ads/aqua/evaluation.py [hunk 2]
```diff
@@ -914,53 +905,105 @@ def download_report(self, eval_id) -> AquaEvalReport:
...
+        Returns
+        -------
+        dict
+            A dict of model evaluation default configs.
+        """
+        target_evaluation = self._get_target_evaluation(
+            target_evaluation_id
+        )
+
+        if target_evaluation.kind == MODEL_DEPLOYMENT_KIND:
```
Comment (mrDzurb) [inline ads/aqua/evaluation.py:932]: NIT: I would prefer to have if/else statement. Returning from if statement makes it a bit difficult to track the logic.

**Review by mrDzurb — APPROVED** [review]

**Comment by lu-ohai** [comment]
> We currently don't have model with `evaluation_config.json` in artifact for our testing. I would suggest to providing a test that proving the response return as expected when `evaluation_config.json` is in artifact.

I guess we can keep this PR and merge it when the required config files are added

**Comment by lu-ohai** [comment]
Closing this PR as a new one has been made to new branch `feature/aquav1.0.2`: https://github.com/oracle/accelerated-data-science/pull/817

**Pull request closed without merging**